e2e/tests/notebook/owner-copy-to-workspace.spec.ts

```diff
@@ -0,0 +1,101 @@
+
+    // Create notebook in copy-from workspace.
+    const copyFromNotebookName = makeRandomName('pytest');
+    const dataPage = new DataPage(page);
+
+    // Get the billing project name from page url.
+    let namespace = extractNamespace(new URL(page.url()));
+
+    let notebookPage = await dataPage.createNotebook(copyFromNotebookName);
+
+    // Run code to print out Workspace namespace.
+    const code =
+       'import os\n' +
+       'print(os.getenv(\'WORKSPACE_NAMESPACE\'))';
+
+    let codeOutput = await notebookPage.runCodeCell(1, {code});
+    expect(codeOutput).toEqual(namespace);
+
+    // Exit notebook and returns to the Workspace Analysis tab.
+    const analysisPage = await notebookPage.goAnalysisPage();
+
+    // Copy to destination Workspace and give notebook a new name.
+    const copiedNotebookName = makeRandomName('copy-of');
+    await analysisPage.copyNotebookToWorkspace(copyFromNotebookName, toWorkspace, copiedNotebookName);
+
+    // Verify Copy Success modal.
+    const modal = new Modal(page);
+    await modal.waitForButton(LinkText.GoToCopiedNotebook);
+    const textContent = await modal.getTextContent();
+    const successMsg = `Successfully copied ${copyFromNotebookName}  to ${toWorkspace} . Do you want to view the copied Notebook?`;
+    expect(textContent).toContain(successMsg);
+    // Dismiss modal.
+    await modal.clickButton(LinkText.StayHere, {waitForClose: true});
+
+    // Delete notebook
+    let modalTextContent = await analysisPage.deleteNotebook(copyFromNotebookName);
+    expect(modalTextContent).toContain(`Are you sure you want to delete Notebook: ${copyFromNotebookName}?`);
+
+    // Perform actions in copied notebook.
+    // Open destination Workspace
+    await findWorkspace(page, {workspaceName: toWorkspace}).then(card => card.clickWorkspaceName());
+    // Get the destination Workspace namespace.
+    namespace = extractNamespace(new URL(page.url()));
+
+    // Verify copy-to notebook exists in destination Workspace
+    await dataPage.openTab(TabLabelAlias.Analysis);
+    const dataResourceCard = new DataResourceCard(page);
+    const notebookCard = await dataResourceCard.findCard(copiedNotebookName, CardType.Notebook);
+    expect(notebookCard).toBeTruthy();
+
+    // Open copied notebook and run code to verify billing project name.
+    await notebookCard.clickResourceName();
+    const notebookPreviewPage = new NotebookPreviewPage(page);
+    await notebookPreviewPage.waitForLoad();
+    notebookPage = await notebookPreviewPage.openEditMode(copiedNotebookName);
+
+    // Run same code and compare namespace.
+    codeOutput = await notebookPage.runCodeCell(-1, {code});
+    expect(codeOutput).toEqual(namespace);
+
+    // Exit notebook. Returns to the Workspace Analysis tab.
+    await notebookPage.goAnalysisPage();
+    // Delete notebook
+    modalTextContent = await analysisPage.deleteNotebook(copiedNotebookName);
```

Comment:
Could these other `let` variables be functions (e.g in this case accept the notebookName as a param)?

Reply:
yeah probably. but it will be more change code. I have other improvement to make in new PR. This would be a good to include in new PR.